VTFLib/VTFFile.cpp

```diff
@@ -3350,6 +3355,74 @@ vlBool CVTFFile::CompressBCn(const vlByte *lpSource, vlByte *lpDest, vlUInt uiWi
 	return vlTrue;
 }
 
+//-----------------------------------------------------------------------------------------------------
+// CompressBC6H()
+// 
+// Compress input image data (lpSource) to output image data (lpDest) of format DestFormat
+// where DestFormat is of format BC6H.  Uses Compressonator library.
+// 
+// NOTE: For BC6H conversion, an extra step is needed that puts the source data into a half float 
+// format before conversion. Do not use BCn for BC6H compression. -klaxon
+//-----------------------------------------------------------------------------------------------------
+vlBool CVTFFile::CompressBC6H(const vlByte* lpSource, vlByte* lpDest, vlUInt uiWidth, vlUInt uiHeight)
+{
+	CMP_Texture srcTexture = { 0 };
+	srcTexture.dwSize = sizeof(srcTexture);
+	srcTexture.dwWidth = uiWidth;
+	srcTexture.dwHeight = uiHeight;
+	srcTexture.dwPitch = 4 * uiWidth;
+	srcTexture.format = CMP_FORMAT_RGBA_8888;
```

Comment:
Is the input to this function always assumed to be RGBA8888?

Reply:
ah, it was while i was testing things out. ill go ahead and add the SourceFormat param back to make sure an edge case isn't hit.

Reply:
✔

Reply:
My issue with this code is less about assuming `lpSource` is always RGBA8888 and more about the potential data loss. If textures are getting squished down to RGBA8888 before being turned into BC6H, what's the point of using BC6H at all

Reply:
This is still blocking merge, basically what JJ said